src/frontend/xgboost_json.cc

```diff
@@ -25,45 +43,363 @@ namespace frontend {
+}
+
+bool XGBoostModelHandler::EndObject(std::size_t memberCount) {
+  if (memberCount != 2) {
+    return false;
+  }
+  m_output.random_forest_flag = false;
+  if (version[0] >= 1) {
+    if (std::strcmp(m_output.param.pred_transform, "sigmoid") == 0) {
+      m_output.param.global_bias =
+          ProbToMargin::Sigmoid(m_output.param.global_bias);
+    } else if (std::strcmp(m_output.param.pred_transform, "exponential") ==
+               0) {
+      m_output.param.global_bias =
+          ProbToMargin::Exponential(m_output.param.global_bias);
+    }
+  }
+  return pop_handler();
+}
+
+/******************************************************************************
+ * RootHandler
+ * ***************************************************************************/
+bool RootHandler::StartObject() {
+  return push_handler<XGBoostModelHandler, treelite::Model>(m_output);
+}
+
+/******************************************************************************
+ * DelegatedHandler
+ * ***************************************************************************/
+treelite::Model DelegatedHandler::get_result() { return std::move(result); }
+bool DelegatedHandler::Null() { return delegates.top()->Null(); }
+bool DelegatedHandler::Bool(bool b) { return delegates.top()->Bool(b); }
+bool DelegatedHandler::Int(int i) { return delegates.top()->Int(i); }
+bool DelegatedHandler::Uint(unsigned u) { return delegates.top()->Uint(u); }
+bool DelegatedHandler::Int64(int64_t i) { return delegates.top()->Int64(i); }
+bool DelegatedHandler::Uint64(uint64_t u) { return delegates.top()->Uint64(u); }
+bool DelegatedHandler::Double(double d) { return delegates.top()->Double(d); }
+bool DelegatedHandler::String(const char *str, std::size_t length, bool copy) {
+  return delegates.top()->String(str, length, copy);
+}
+bool DelegatedHandler::StartObject() { return delegates.top()->StartObject(); }
+bool DelegatedHandler::Key(const char *str, std::size_t length, bool copy) {
+  return delegates.top()->Key(str, length, copy);
+}
+bool DelegatedHandler::EndObject(std::size_t memberCount) {
+  return delegates.top()->EndObject(memberCount);
+}
+bool DelegatedHandler::StartArray() { return delegates.top()->StartArray(); }
+bool DelegatedHandler::EndArray(std::size_t elementCount) {
+  return delegates.top()->EndArray(elementCount);
+}
+
+
+}  // namespace details
+}  // namespace treelite
+
+namespace {
+template<typename StreamType>
+treelite::Model ParseStream(std::unique_ptr<StreamType> input_stream) {
+  std::shared_ptr<treelite::details::DelegatedHandler> handler = \
+    treelite::details::DelegatedHandler::create();
+  rapidjson::Reader reader;
+
+  rapidjson::ParseResult result = reader.Parse(*input_stream, *handler);
+  if (!result) {
+    LOG(ERROR) << "Parsing error " << result.Code() << " at offset "
+               << result.Offset();
+    throw std::runtime_error(
+        "Provided JSON could not be parsed as XGBoost model");
```

Comment:
```suggestion
    LOG(FATAL) << "Provided JSON could not be parsed as XGBoost model";
```
Let's use the DMLC logger here to be consistent with other parts of Treelite. (Note. The `tree.h` header uses a standard exception because it cannot include the DMLC header.)